### DISCUSSION TITLE
How to determine if the document does not exist at all or it's indeed still 'loading'?

### DISCUSSION THREAD

**Discussion by namiwang** [FirebaseExtended/reactfire#438] [Q&A] [answered]
Take the firestore example here (https://stackblitz.com/edit/reactfire-v4-sample), if we change the querying doc path to

```
const burritoRef = doc(useFirestore(), 'tryreactfire', 'noburrito');
```

and it will be stuck in 'fetching burrito flavor' forever.

In some cases, we'd like to determine the request is finished and the document doesn't exist for now. Is there any way we could detect such a condition in reactfire? Thanks!

**Comment by jhuleatt** [comment, marked as answer]
Hi @namiwang, in that case the `data` field of the object returned by `useFirestoreDocData` will be `undefined`, and `status` will be `success`.  This means that ReactFire is actively listening for that document in Firestore, but nothing is there. So you can just add a check:

```jsx
function BurritoTaste() {
  // easily access the Firestore library
  const burritoRef = doc(useFirestore(), 'tryreactfire', 'burrito');

  // subscribe to a document for realtime updates. just one line!
  const { status, data } = useFirestoreDocData(burritoRef);

  // easily check the loading status
  if (status === 'loading') {
    return <p>Fetching burrito flavor...</p>;
  } else if (data === undefined) {
    return <p>Burrito not found!</p>;
  }

  return <p>The burrito is {data.yummy ? 'good' : 'bad'}!</p>;
}
```

**Reply by namiwang** [reply]
Hi @jhuleatt , thanks for the reply.

The behavior you described is just what I expected, yet in fact, it stucked on `loading` forever, never went to `success`.

Here's a forked demo (https://stackblitz.com/edit/reactfire-v4-sample-gguyd1)

<img width="1230" alt="Screen Shot 2021-08-30 at 11 37 17" src="https://user-images.githubusercontent.com/1160688/131281693-f065844a-3a34-4ca1-bd73-d2f7576788aa.png">

**Reply by namiwang** [reply]
BTW I got what I want (`success` with `undefined`) using `useDocDataOnce` instead of `useDocData`, but the issue remains

**Reply by jhuleatt** [reply]
Thanks for the additional detail, @namiwang. This seems like a bug. I've reported it in https://github.com/FirebaseExtended/reactfire/issues/440

**Reply by atanaskanchev** [reply]
+1 I can confirm the same behaviour with version ```4.1.0```

**Reply by jhuleatt** [reply]
This turned out to be a bug in the core of ReactFire (`useObservable`, the hook that all other ReactFire hooks call). I'm working on a fix in #446 if anyone is interested in giving it a review. Thanks for bringing this to my attention!

**Reply by jhuleatt** [reply]
fixed in ReactFire [4.1.1](https://github.com/FirebaseExtended/reactfire/releases/tag/v4.1.1)
